skbio/sequence/_iupac_sequence.py

```diff
@@ -164,23 +165,72 @@ def _motifs(self):
 
     @overrides(Sequence)
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 validate=True, case_insensitive=False):
+                 validate=True, lowercase=False):
         super(IUPACSequence, self).__init__(
             sequence, metadata, positional_metadata)
 
-        if case_insensitive:
+        # Note we are checking identity against the False object here and not
+        # just falsiness
+        if lowercase is not False:
+            self._lowercase_mask = self._bytes > self._ascii_lowercase_boundary
```

Comment:
Can you refactor this so that `_lowercase_mask` isn't being stored as an instance variable? We want users to be able to say `lowercase=True` and not have lowercase info tracked on a sequence object (e.g., for memory efficiency).